pkg/my-sql/adapter_test.go

```diff
@@ -496,6 +496,200 @@ func TestAdapterGetPage(t *testing.T) {
 				},
 			},
 		},
+		"validation_prevent_sql_injection_unique_attribute": {
+			request: &framework.Request[mysql.Config]{
+				Address: "127.0.0.1",
+				Auth: &framework.DatasourceAuthCredentials{
+					Basic: &framework.BasicAuthCredentials{
+						Username: "testusername",
+						Password: "testpassword",
+					},
+				},
+				Entity: framework.EntityConfig{
+					ExternalId: "Users",
+					Attributes: []*framework.AttributeConfig{
+						{
+							ExternalId: "DROP sampletable;-- ",
+							Type:       framework.AttributeTypeString,
+							UniqueId:   true,
+						},
+					},
+				},
+				Config: &mysql.Config{
+					CommonConfig: &config.CommonConfig{
+						RequestTimeoutSeconds: testutil.GenPtr(10),
+						LocalTimeZoneOffset:   -18000, // UTC−05:00 (EST)
+					},
+					Database: "sgnl",
+				},
+				Ordered:  true,
+				PageSize: 5,
+				Cursor:   "10",
+			},
+			wantResponse: framework.Response{
+				Error: &framework.Error{
+					Message: "SQL unique attribute validation failed: unsupported characters found.",
+					Code:    api_adapter_v1.ErrorCode_ERROR_CODE_INVALID_ENTITY_CONFIG,
+				},
+			},
+		},
+		"valid_request_first_page_cast_multiple_fields_to_string": {
```

Comment:
As we can see in this test, a user that now requests an attribute of type `String` will result in a `String` being ingested regardless of the type stored in the SQL database. We can compare this to the `valid_request_first_page` test case (first test case in this map) to see the behavior when a user specifies the type that matches the type in SQL.